test/robot/go_robot_go.sh

```diff
@@ -0,0 +1,320 @@
+#!/bin/bash
+shopt -s extglob
+clear
+pushd $(dirname $0/../..) > /dev/null
+OUTDIR="bazel-out/robot/"
+BUILD_BOT_DIR="${OUTDIR}build_bot/"
+
+function handle_opts() {
+  while true; do
+    case $1 in
+      -c|--clean) CLEAN=1; shift ;;
+      --) shift; break ;;
+      *) echo "internal error!" ; exit 1 ;;
+    esac
+  done
+}
+handle_opts $(getopt -l "clean" -o "c" -n "go_robot_go.sh" -- "$@")
+
+function howmany() { echo $#; }
+
+function find_pid() { pgrep -f "$1" -d " "; }
+
+function is_device_awake() {
+  STATE=`adb -s $1 shell dumpsys nfc \
+    | grep mScreenState \
+    | awk '{split($1,a,"="); print a[2]}'`
+
+  if [[ "$STATE" = "ON_UNLOCKED" ]]; then
```

Comment:
`return [[ "$STATE" = "ON_UNLOCKED" ]]`

and then later on

`if is_device_awake $device; then ...`